### DISCUSSION TITLE
How to delete a file from a volume that's cloud hosted?

### DISCUSSION THREAD

**Discussion by tobiasmuehl** [seaweedfs/seaweedfs#5513] [Q&A] [answered]
Is there a way to delete a file from a volume that has already been moved to the warm cache tier on S3 or equivalent? Will seaweedfs need to download the whole bundle onto a volume server to do the edit? 

The context of this is GDPR data deletion requests, so it's important that all copies of the file are deleted

**Comment by chrislusf** [comment]
there is a soft delete marker set when deleting an entry in a cloud volume.

**Reply by tobiasmuehl** [reply]
What does that mean specifically? For GDPR compliance we need to ensure the data is actually deleted, not just marked as deleted. Does Soft Delete just apply a flag in Filer?

**Reply by chrislusf** [reply, marked as answer]
It's soft delete. To actually delete the blob, need to pull down the volume from the cloud tier, run compaction, and maybe upload back the volume.

The reference to the blob is forgotten, which should be good enough for GDPR.
